src/main/java/com/google/devtools/build/lib/exec/ExecutionOptions.java

```diff
@@ -118,6 +121,31 @@ public class ExecutionOptions extends OptionsBase {
               + "the 'local' strategy, but reversing the order would run it with 'sandboxed'. ")
   public List<Map.Entry<RegexFilter, List<String>>> strategyByRegexp;
 
+  @Option(
+      name = "allowed_strategies_by_exec_platform",
+      allowMultiple = true,
+      converter = LabelToStringListConverter.class,
+      defaultValue = "null",
+      documentationCategory = OptionDocumentationCategory.EXECUTION_STRATEGY,
+      effectTags = {OptionEffectTag.EXECUTION},
+      help =
+          """
+          Filters spawn strategies by the execution platform without affecting order.
+          For example:
+          ```
+          common --spawn_strategy=remote,sandboxed,worker,local
+          common --strategy=Genrule=local
+          common --allowed_strategies_by_exec_platform=@platforms//host:host=local,sandboxed,worker
+          common --allowed_strategies_by_exec_platform=//:linux_amd64=remote
+          ```
+          With the above options;
+          - Actions configured for the host platform will be given `remote,sandboxed,worker`.
+          - Actions configured for the `//:linux_amd64` platform will be given `remote`.
+          - Actions configured for the `//:linux_amd64` platform with mnemonic `Genrule` will be
+            given no strategies.
```

Comment:
Does "given no strategies" then cause an error? What happens in that case?

Reply:
Docs updated to clarify that spawn will fail.